WorldModel/WorldModel/Core/CoreTypes.aslx

```diff
@@ -323,6 +323,7 @@
     <hasbeenmoved type="boolean">false</hasbeenmoved>
     <timesexamined type="int">0</timesexamined>
     <not_all type="boolean">false</not_all>
+    <isroom/>

```

Comment:
This is the only thing I'm a bit unsure about. If you weren't using the editor, that would mean all objects would appear in `AllRooms`? It seems like it would make more sense if this were to default to false (or be undefined).

Reply:
That's how I tried it at first, but changing to "object and/or room" kept removing `isroom` because it removes `editor_object` and `editor_room`. The editor controls work "backwards". The room controls make sure it doesn't have the `editor_object` type, and vice-versa.

So, Pertex thought backwards, and it works. It is very counter-intuitive, though. And apparently [having the code in a different order messes it up](https://github.com/textadventures/quest/pull/1324#discussion_r1899955538), unless I am missing something (which is very likely).

> If you weren't using the editor, that would mean all objects would appear in AllRooms?

Well, technically, we moved those types to Core.aslx, so this should still work even if I used a text editor and zipped it up. I know you know that, though, and I see what you mean. I don't like it either.

> It seems like it would make more sense if this were to default to false (or be undefined).

I've changed something in Pertex's code somewhere, because he couldn't return all the rooms correctly without modifying the script to check for objects *without* `isroom` and objects with it set to `true`. So, that sounds like he had it like you are describing it. Let me look at his original code again.

Reply:
Yes, code generated object would be rooms. Thats bad but I am checking a new idea.

Reply:
... No, all I see different is the order is ever so slightly different.

---
Pertex #1316:
![image](https://github.com/user-attachments/assets/3a511e90-d266-4aad-97aa-64df5c6194d2)

---
This PR:
![image](https://github.com/user-attachments/assets/ed67ec83-dd0c-48ca-b185-93b7ba8e487f)

---
And our `default_object` and `default_room` bits are inverted:

---
Pertex #1316:
![image](https://github.com/user-attachments/assets/60e83257-9673-46a3-9d19-d0ceb1844957)

---
This PR:
![image](https://github.com/user-attachments/assets/c164c60e-6f23-411b-9486-4b8ca91e3604)

---
I didn't switch things around on purpose, either. I just usually try to add new stuff to the bottom, unless it should be grouped near something.

Anyway, after this PR was submitted, I looked at his code again and noticed he'd changed `AllRooms`. He's looping through `AllObjects` and filtering like this in his old PR:
```cs
if (HasBoolean (o, "isroom") = false or GetBoolean (o, "isroom") = true){
```

He said he had to change that because his `"object and/or room"` objects don't make the list (because `isroom` is undefined with his code), but `isroom` gets set to `true` in these instances the way I did it -- and I didn't modify `AllRooms` (simply because I overlooked it in his commits). I even toggled things in the drop-down purposefully and named them accordingly for easy testing in the attached game.

I'm rambling on because I don't quite grasp why it doesn't work the same with the code in a slightly different order.

| type | `isroom` |
|-|-|
| `defaultobject` | true |
| `editor_object` | false |
| `editor_room` | true |

If the editor is set on "Object", `editor_object` overrides `defaultobject`, making `isroom` false.

Making it a "Room" makes `isroom` true, and making it "Object and/or room" removes everything but `defaultobject`, leaving it true. Okay, I see why my order made a difference. In his code, `editor_object` has the last say, but mine is reversed. 🤯 

Sorry... Had to talk that out to get my head around it!

---
>  If you weren't using the editor, that would mean all objects would appear in `AllRooms`?

I didn't think this through. Why would someone mess with the `editor_*` types if they weren't using the editor (unless they were made aware of code such as this)?

I am by no means trying to convince you to keep this backwards if you don't want to. What about switching the editor controls around? Specifically this bit: `<mustnotinherit>editor_room</mustnotinherit>`

Reply:
OK, the real problem is that there is no type for “object and room”
       <control>
         <caption>[EditorObjectSetupType]</caption>
         <controltype>dropdowntypes</controltype>
        <types>editor_room=[TypeObjectRoom]; editor_object=[TypeObjectObject]; *=[TypeObjectAndOr]</types>
        <width>150</width>
      <advanced/>
 </control>
 
 Now I wanted to create a new Editor_type
 
<types>editor_room=[TypeObjectRoom]; editor_object=[TypeObjectObject]; editor_room_object=[TypeObjectAndOr]</types>

This would allow you to specify for each type whether it is a room, but in existing games this causes the error that * does not exist. I have two ideas for this. The first: Set such objects to editor_room_object when loading. However, I don't know if and how this is feasible. 
The second idea: create a new type "Undefined"

<types>editor_room=[TypeObjectRoom]; editor_object=[TypeObjectObject]; editor_room_object=[TypeObjectAndOr]; *=[TypeObjectUndefined]</types>

Objects created by code would be undefined and treated like previous “object and room” objects, except that they are not rooms
Now the question is whether this makes sense or is it overdosed?

Reply:
I think that's a good idea. The way things currently work, with "object and/or room" having _no_ type, means you only have `defaultobject`, but we do want "object and/or room" to has `isroom` set.

"Undefined" at least should be pretty uncommon. So I say let's go ahead with adding an `editor_room_object` type.

Reply:
Pertex sent me revised files, and the changes have been pushed (and applied to this PR).

We have decided to change the value of the `*=[TypeObjectUndefined]` template from "Object is undefined" to "Object type not specified". We thought we'd run that past you first, though, @alexwarren

Reply:
On second thought, how about that template for `TypeObjectUndefined` is just "Object"?

Plain. Simple. True.

Reply:
We just changed it to 'Objekttyp ist nicht festgelegt'

... I mean 'Object type is not set'